src/team.hpp

```diff
@@ -264,31 +258,35 @@ class team
 		}
 	}
 
-	CONTROLLER controller() const { return info_.controller; }
+	side_controller::TYPE controller() const { return info_.controller; }
 	const std::string& color() const { return info_.color; }
 	/** @note Call display::reinit_flag_for_side() after calling this */
 	void set_color(const std::string& color) { info_.color = color; }
-	bool is_empty() const { return info_.controller == CONTROLLER::EMPTY; }
+	bool is_empty() const { return info_.controller == side_controller::TYPE::NONE; }
 
 	bool is_local() const { return !is_empty() && info_.is_local; }
 	bool is_network() const { return !is_empty() && !info_.is_local; }
 
-	bool is_human() const { return info_.controller == CONTROLLER::HUMAN; }
-	bool is_ai() const { return info_.controller == CONTROLLER::AI; }
+	bool is_human() const { return info_.controller == side_controller::TYPE::HUMAN; }
+	bool is_ai() const { return info_.controller == side_controller::TYPE::AI; }
 
 	bool is_local_human() const {  return is_human() && is_local(); }
 	bool is_local_ai() const { return is_ai() && is_local(); }
 	bool is_network_human() const { return is_human() && is_network(); }
 	bool is_network_ai() const { return is_ai() && is_network(); }
 
 	void set_local(bool local) { info_.is_local = local; }
-	void make_human() { info_.controller = CONTROLLER::HUMAN; }
-	void make_ai() { info_.controller = CONTROLLER::AI; }
+	void make_human() { info_.controller = side_controller::TYPE::HUMAN; }
+	void make_ai() { info_.controller = side_controller::TYPE::AI; }
 	void change_controller(const std::string& new_controller) {
-		info_.controller = CONTROLLER::AI;
-		info_.controller.parse(new_controller);
+		std::optional<side_controller::TYPE> type = side_controller::get_enum(new_controller);
```

Comment:
This (and other similar cases) is a perfect use for `optional::value_or`, since the optional will be a nullopt if `new_controller` is invalid:
```cpp
info_.controller = side_controller::get_enum(new_controller).value_or(side_controller::TYPE::AI);
```

Reply:
Done.